agent/src/api/v1/jobs.rs

```diff
@@ -54,15 +54,25 @@ async fn run_job(
     state: State<AppState>,
     input: Bytes,
 ) -> impl IntoResponse {
-    let storage = STORAGE.get().expect("Storage not initialized!");
+    let Ok(storage) = crate::storage::get_runner_storage().await else {
+        return (StatusCode::SERVICE_UNAVAILABLE, "unable to locate a storage node on the mesh".to_string()).into_response();
+    };
+
     let Ok(manifest) = storage.manifest(&name).await else {
         return (StatusCode::NOT_FOUND, "no manifest of that name found").into_response();
     };
 
     let Ok(executable) = storage.executable_as_bytes(&name, manifest.version()).await else {
-        return (StatusCode::NOT_FOUND, "no executable found for manifest; key={key}").into_response();
+        return (StatusCode::NOT_FOUND,
+            format!("no executable found for manifest;  name={name}; version={}", manifest.version())).into_response();
     };
 
+    if executable.is_empty() {
+        let warning = format!("Declining to run a job of zero length; name={}; version={}", &name, manifest.version());
+        log::warn!("{warning}");
```

Comment:
Nit, but `log::warn!` accepts format arguments so you don't actually need to do the `let warning` bit